packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -124,144 +129,156 @@ export const render = async (req, res, next) => {
-            __STATE_MANAGEMENT_LIBRARY: AppConfig.freeze(res.locals)
+    // Auto-enable OpenTelemetry and B3 tracing when performance tracking is requested
+    if (shouldTrackPerformance) {
+        if (!process.env.OTEL_SDK_ENABLED) {
+            process.env.OTEL_SDK_ENABLED = 'true'
         }
-        appStateError = ret.error
-        res.__performanceTimer.mark(PERFORMANCE_MARKS.fetchStrategies, 'end')
-    }
-    res.__performanceTimer.mark(PERFORMANCE_MARKS.renderToString, 'start')
-    appJSX = React.cloneElement(appJSX, {error: appStateError, appState})
-
-    // Step 4 - Render the App
-    let renderResult
-    try {
-        renderResult = renderApp({
-            App: WrappedApp,
-            appState,
-            appStateError: appStateError && logAndFormatError(appStateError),
-            routes,
-            req,
-            res,
-            location,
-            config,
-            appJSX
-        })
-    } catch (e) {
-        // This is an unrecoverable error.
-        // (errors handled by the AppErrorBoundary are considered recoverable)
-        // Here, we use Express's convention to invoke error middleware.
-        // Note, we don't have an error handling middleware yet! This is calling the
-        // default error handling middleware provided by Express
-
-        if (res.__performanceTimer) {
-            res.__performanceTimer.cleanup()
+        if (!process.env.OTEL_B3_TRACING_ENABLED) {
+            process.env.OTEL_B3_TRACING_ENABLED = 'true'
         }
-
-        return next(e)
     }
 
-    // Step 5 - Determine what is going to happen, redirect, or send html with
-    // the correct status code.
-    const {html, routerContext, error} = renderResult
-    const redirectUrl = routerContext.url
-    const status = (error && error.status) || res.statusCode
-
-    res.__performanceTimer.mark(PERFORMANCE_MARKS.renderToString, 'end')
-    res.__performanceTimer.mark(PERFORMANCE_MARKS.total, 'end')
-    res.__performanceTimer.log()
-
-    if (includeServerTimingHeader) {
-        res.setHeader('Server-Timing', res.__performanceTimer.buildServerTimingHeader())
-
-        // Override cache-control header to no caching when __server_timing is used
-        // This happens after React rendering is complete, ensuring it overrides any
-        // cache headers set by individual page components
-        res.set('Cache-Control', NO_CACHE)
+    // Initialize server tracing if needed
+    if (shouldTrackPerformance && !isServerTracingInitialized()) {
+        initializeServerTracing()
     }
 
-    res.__performanceTimer.cleanup()
-
-    if (redirectUrl) {
-        res.redirect(routerContext.status || 302, redirectUrl)
-    } else {
-        res.status(status).send(html)
-    }
+    return tracePerformance(
+        'ssr.render',
+        async () => {
+            res.__performanceTimer = new PerformanceTimer({enabled: shouldTrackPerformance})
+            res.__performanceTimer.mark(PERFORMANCE_MARKS.total, 'start')
```

Comment:
There are two ways of marking performance, the new `tracePerformance` and the previously implemented `res.__performanceTimer`, can we consolidate them into a single interface?